cirq-core/cirq/sim/density_matrix_simulator_test.py

```diff
@@ -1133,13 +1133,15 @@ def test_density_matrix_trial_result_qid_shape():
 
 def test_density_matrix_trial_result_repr():
     q0 = cirq.LineQubit(0)
+    dtype = np.complex64
```

Comment:
If we leave this as `np.float64` does the repr stay as it was?

Reply:
Oh, never mind, it looks like basically the old test was unrealistic because it passed in a `float64` which would never occur in the real world (pun intended) anyway. This looks good.

Reply:
Thanks. Previously, `target_tensor` was assigned in `ActOnDensityMatrixArgs.__init__` without checking. After adding the logic in `density_matrix_simulator._create_partial_act_on`, `float64` lead to test errors.